Code/src/main/java/nl/utwente/viskell/ui/components/ChoiceBlock.java

```diff
@@ -0,0 +1,140 @@
+package nl.utwente.viskell.ui.components;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javafx.fxml.FXML;
+import javafx.scene.layout.Pane;
+import nl.utwente.viskell.haskell.expr.Binder;
+import nl.utwente.viskell.haskell.expr.Case;
+import nl.utwente.viskell.haskell.expr.Case.Alternative;
+import nl.utwente.viskell.haskell.expr.Expression;
+import nl.utwente.viskell.haskell.expr.Value;
+import nl.utwente.viskell.haskell.type.HaskellTypeError;
+import nl.utwente.viskell.haskell.type.Type;
+import nl.utwente.viskell.haskell.type.TypeChecker;
+import nl.utwente.viskell.haskell.type.TypeScope;
+import nl.utwente.viskell.haskell.type.TypeVar;
+import nl.utwente.viskell.ui.CustomUIPane;
+
+/**
+ * An evaluation block with multiple guarded alternatives.
+ *
+ */
+public class ChoiceBlock extends Block {
+    
+    /** The alternatives inside this block */
+    protected List<Lane> lanes;
+    
+    /** The output anchor of this block */
+    protected OutputAnchor output;
+    
+    /** The container Node for the Lanes */
+    @FXML protected Pane altSpace;
+    
+    /** The container Node for the OutputAnchor */
+    @FXML protected Pane funSpace;
```

Comment:
Yes, because the result of a DefinitionBlock is always a function ('hence the fun'). The result of a ChoiceBlock could be any value, right?